### PR TITLE
remove old cruft, use selector

### PR DESCRIPTION
**Pull request by braver**

### CODE REVIEW

File: linter.py [hunk 1]
```diff
@@ -1,26 +1,14 @@
...
-
-    syntax = ('go', 'gosublime-go')
     executable = "gotype"
     regex = r'(?P<filename>^.+):(?P<line>\d+):(?P<col>\d+):\s+(?P<message>.+)'
     error_stream = util.STREAM_STDERR
+    defaults = {
+        'selector': 'source.go'
+    }
 
     def __init__(self, view, syntax):
```
Comment (kaste) [inline linter.py]: Remove __init__ completely, user can set env at will.

executable and executable_path are deprecated

Comment (braver) [reply linter.py]: Like so? 5ed938b

File: linter.py [hunk 2]
```diff
@@ -1,41 +1,13 @@
...
-from os.path import dirname
 from SublimeLinter.lint import Linter, util
 
 
 class Gotype(Linter):
-    """Provides an interface to gotype."""
-
-    syntax = ('go', 'gosublime-go')
-    executable = "gotype"
+    cmd = ('gotype', '-e', '${file}')
```
Comment (kaste) [inline linter.py]: $file_path here bc below it was dirname()

**Comment by braver** [comment]
I guess the gopath setting stops working with this PR..

File: README.md [hunk 3]
```diff
@@ -3,42 +3,36 @@ SublimeLinter-gotype
...
+- Linter settings: http://sublimelinter.com/en/latest/linter_settings.html
 
-|Setting|Description         |Inline Setting|Inline Override|
-|:------|:-------------------|:------------:|:-------------:|
-|gopath |Set a custom GOPATH |||
+Additional SublimeLinter-gotype provides settings:
+
+|Setting|Description         |
+|:------|:-------------------|
+|gopath |Set a custom GOPATH |
```
Comment (rasky) [inline README.md]: Is this still available? It looks like it's removed with your PR.

**Comment by kaste** [comment]
This wasn't working before (SL4 broke this linter) and using the env setting is the future.

**Comment by rasky** [comment]
@braver @kaste this doesn't seem to work, I get this error:

```
SublimeLinter: #20 SublimeLinter.sublime-package:814: gotype: No match for line: 'stat ${file_path}: no such file or directory'
```

File: README.md (hunk 3 above)
Comment (braver) [reply README.md]: Indeed, it’s gone. All linters take an “env” setting where you can set things like this.

**Comment by kaste** [comment]
@rasky That `stat ${file_path}` has been fixed, sorry.

**Pull request merged**